File: 238_Product_of_Array_Except_Self/main.go
func productExceptSelf(nums []int) []int {
	l := len(nums)
	ans := []int{1}

	for i := 1; i < l; i++ {
		ans = append(ans, ans[i-1]*nums[i-1])
	}
	fmt.Println(ans)
	postfixProduct := 1
	for i := l - 1; i >= 0; i-- {
		ans[i] *= postfixProduct
		postfixProduct *= nums[i]
	}

	return ans
}

File: 238_Product_of_Array_Except_Self/main.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l = len(nums)
        ans = [1]*l
        for i in range(1,l):
            ans[i] = ans[i-1] * nums[i-1]
        postfixProd = 1
        for i in range(l-1, -1, -1):
            ans[i] *= postfixProd
            postfixProd *= nums[i]

        return ans
